### ISSUE TITLE
Rename default branch for kubernetes/k8s.io

### ISSUE THREAD

**Issue by spiffxp** [kubernetes/k8s.io#1597]
Part of umbrella issue https://github.com/kubernetes/org/issues/2222

Issue to spec out and then run through renaming the default branch for this repository

- update pre/postsubmits to trigger on old branch name and new branch name
- determine whether periodics use default branch or have name hardcoded in git commands
  - if hardcoded, does attempting to checkout old name break?
- determine whether branch name is part of any job names
- determine if any docs need to be updated
- send notice
- rename branch (ref: https://github.com/github/renaming#renaming-existing-branches)
- if necessary, update periodics to refer to new branch name
- if necessary, rename jobs
- update branch protection config
- update jobs to only trigger on new branch name
- make sure things still work
  - keep an eye on ci and prs for a bit
  - trial the local development experience with pre-rename clone
- send notice

/sig contributor-experience
/area github-management
trialing a repo for this subproject
/wg k8s-infra
owns this repo
/sig testing
to discover any prow issues
/milestone v1.21
/priority important-soon
/assign @spiffxp 
I plan on speccing this out, whether/when to execute depends on how kubernetes/org goes

**Comment by spiffxp** [comment]
PR up for job config changes + prow config changes: https://github.com/kubernetes/test-infra/pull/20675 (holding merge until next week, and completion of the below review)

Still need to:
- review the code in k8s.io to confirm whether it explicitly references the master branch...
- review docs to see whether master is explicitly references (beyond uris that github will redirect for us)

**Comment by spiffxp** [comment]
Merged https://github.com/kubernetes/test-infra/pull/20675

**Comment by spiffxp** [comment]
Renamed the branch

**Comment by spiffxp** [comment]
Will leave as-is and cleanup tomorrow

Known issues:
- go.k8s.io/redirects 404's for me (but visiting https://github.com/kubernetes/k8s.io/tree/master/k8s.io#redirections is properly redirected to main by github)
https://github.com/kubernetes/k8s.io/blob/a8e0b654626c77de30220097cf8253d32363bcb8/k8s.io/configmap-nginx.yaml#L242

**Comment by spiffxp** [comment]
looking at jobs in our test grid:
- [ ] https://testgrid.k8s.io/wg-k8s-infra-k8sio#ci-k8sio-groups
- [x] https://testgrid.k8s.io/wg-k8s-infra-k8sio#ci-k8sio-image-promo
- [ ] https://testgrid.k8s.io/wg-k8s-infra-k8sio#ci-release-vulndash-update
- [x] https://testgrid.k8s.io/wg-k8s-infra-k8sio#post-k8sio-deploy-prow-build-resources
- [ ] https://testgrid.k8s.io/wg-k8s-infra-k8sio#post-k8sio-deploy-prow-build-trusted-resources (no recent runs)
- [ ] https://testgrid.k8s.io/wg-k8s-infra-k8sio#post-k8sio-dns-update (no recent runs)
- [ ] https://testgrid.k8s.io/wg-k8s-infra-k8sio#post-k8sio-groups
- [x] https://testgrid.k8s.io/wg-k8s-infra-k8sio#post-k8sio-image-promo
- [ ] https://testgrid.k8s.io/wg-k8s-infra-k8sio#post-k8sio-infra-tools-push-images (no recent runs)
- [ ] https://testgrid.k8s.io/wg-k8s-infra-k8sio#pull-k8sio-backup (no recent runs)
- [x] https://testgrid.k8s.io/wg-k8s-infra-k8sio#pull-k8sio-cip
- [ ] https://testgrid.k8s.io/wg-k8s-infra-k8sio#pull-k8sio-cip-vuln (no recent runs)
- [ ] https://testgrid.k8s.io/wg-k8s-infra-k8sio#pull-k8sio-groups-test

**Comment by spiffxp** [comment]
https://kubernetes.slack.com/archives/C1TU9EB9S/p1613578896224800 - @xmudrii noticed cs.k8s.io wasn't able to search this repo anymore, @dims enabled `detect-ref` and we're searchable again

**Comment by spiffxp** [comment]
I'm going to assume the jobs that have no recent runs are fine given everything else worked

**Comment by spiffxp** [comment]
Looking for references to `master` within this repo:
- I think we have cip-auditor in cloud run hardcoded to look at `master`, see https://github.com/kubernetes/k8s.io/blob/4cf9780327d115ef4c2a80a201bca02556a60bf7/infra/gcp/cip-auditor/deploy.sh#L49
  - but also, looking at logs for it, I see a lot of 500's, I think something else is broken with it
  - manually deployed a new revision with the new branch name ![Screen Shot 2021-02-17 at 7 42 40 PM](https://user-images.githubusercontent.com/49258/108287181-66c24300-7158-11eb-8631-d2ec5479ff13.png)
- dns/octodns-docker/cloudbuild.yaml has 'master' as a default
- there are some links to k8s.io.*master (github would redirect those anyway, but we can clean them up)

**Comment by spiffxp** [comment]
Still need to cleanup references to master in this repo

**Blocking**
- [`cip-auditor`][1]: this is on me, is now blocked behind https://github.com/kubernetes/k8s.io/issues/1730 otherwise I would ask releng
- [`go.k8s.io/redirects`][2] - Can be PR'ed by anyone. Approval should be restricted to someone who can deploy the `k8s.io` app.  I don't have experience there, would prefer someone else can unblock.<sup>1</sup>  

**Nice to have**
- [`dns/octodns-docker/cloudbuild.yaml`][3]: Can be PR'ed by anyone.  Should auto-deploy on merge.
- [clean up links to `k8s.io/.*/(blob|tree)/master`][4]: Can be PR'ed by anyone, approved by relevant subproject owners or approvers.


<sup>1</sup> I would like to learn how to deploy k8s.io, but it may be a while before I can get there.

[1]: https://github.com/kubernetes/k8s.io/blob/4cf9780327d115ef4c2a80a201bca02556a60bf7/infra/gcp/cip-auditor/deploy.sh#L49
[2]: https://github.com/kubernetes/k8s.io/blob/a8e0b654626c77de30220097cf8253d32363bcb8/k8s.io/configmap-nginx.yaml#L242
[3]: https://github.com/kubernetes/k8s.io/blob/main/dns/octodns-docker/cloudbuild.yaml
[4]: https://cs.k8s.io/?q=%2Fk8s.io%2F.*(tree%7Cblob)%2Fmaster%2F&i=nope&files=&excludeFiles=&repos=

**Comment by spiffxp** [comment]
@nikhita and @dims have helped land some of the necessary changes, edited above comment into checklist and crossed out what's been done

**Comment by spiffxp** [comment]
/help

We could use some help closing out "clean up links to `k8s.io/(blob|tree)/master`" above.  A PR per repo would be fine

List generated via:
```bash
curl -s -X POST \
  -F "q=/k8s.io/(tree|blob)/master/" \
  -F "repos=*" \
  "https://cs.k8s.io/api/v1/search" \
| jq -r '.Results | keys | .[]' \
| sort \
| sed -e 's/^/- [ ] /g'
```
- [ ] kubernetes-csi/csi-driver-host-path
- [ ] kubernetes-csi/csi-driver-iscsi
- [ ] kubernetes-csi/csi-driver-nfs
- [ ] kubernetes-csi/csi-lib-utils
- [ ] kubernetes-csi/csi-proxy
- [ ] kubernetes-csi/csi-release-tools
- [ ] kubernetes-csi/csi-test
- [ ] kubernetes-csi/external-attacher
- [ ] kubernetes-csi/external-health-monitor
- [ ] kubernetes-csi/external-provisioner
- [ ] kubernetes-csi/external-resizer
- [ ] kubernetes-csi/external-snapshotter
- [ ] kubernetes-csi/livenessprobe
- [ ] kubernetes-csi/node-driver-registrar
- [ ] kubernetes-csi/volume-data-source-validator
- [ ] kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm
- [ ] kubernetes-sigs/cluster-api-provider-aws
- [ ] kubernetes-sigs/cluster-api-provider-digitalocean
- [ ] kubernetes-sigs/cluster-api-provider-openstack
- [ ] kubernetes-sigs/descheduler
- [ ] kubernetes-sigs/gcp-compute-persistent-disk-csi-driver
- [ ] kubernetes-sigs/gcp-filestore-csi-driver
- [ ] kubernetes-sigs/k8s-container-image-promoter
- [ ] kubernetes-sigs/metrics-server
- [ ] kubernetes-sigs/nfs-ganesha-server-and-external-provisioner
- [ ] kubernetes-sigs/nfs-subdir-external-provisioner
- [ ] kubernetes-sigs/scheduler-plugins
- [ ] kubernetes-sigs/secrets-store-csi-driver
- [ ] kubernetes/community
- [ ] kubernetes/dns
- [ ] kubernetes/ingress-nginx
- [ ] kubernetes/k8s.io
- [ ] kubernetes/kubernetes
- [ ] kubernetes/perf-tests
- [ ] kubernetes/publishing-bot
- [ ] kubernetes/sig-release
- [ ] kubernetes/test-infra
- [ ] kubernetes/website

**Comment by praveenghuge** [comment]
I can take up the kubernetes-sigs repos

**Comment by anubha-v-ardhan** [comment]
I can take up all the kubernetes-csi/ repos

**Comment by RinkiyaKeDad** [comment]
I'll take the remaining kubernetes repos then :)

/assign

**Comment by anubha-v-ardhan** [comment]
Done all the kubernetes-csi/ repos : )

**Comment by anubha-v-ardhan** [comment]
Closed all my PRs except https://github.com/kubernetes-csi/csi-release-tools/pull/164 as stated in this comment https://github.com/kubernetes-csi/csi-release-tools/pull/164#issuecomment-872097531

**Comment by spiffxp** [comment]
/assign @anubha-v-ardhan @praveenghuge @RinkiyaKeDad
Thank you for taking this on!  I've updated https://github.com/kubernetes/k8s.io/issues/1597#issuecomment-871796515 based on latest results

Sounds like @pohly will help out with the remainder of the kubernetes-csi repos

**Comment by RinkiyaKeDad** [comment]
All the PRs for `kubernetes/` have now been merged @spiffxp. Please let us know if there is something else we could help with :D

**Comment by spiffxp** [comment]
I refreshed https://github.com/kubernetes/k8s.io/issues/1597#issuecomment-871796515 and poked the open PRs I could find.  There are still some outstanding repos I couldn't find a linked repo for.

Leaving this in v1.22 because it may be basically done if we can figure out how many repos are blocked by the "csi tools will be copied" reasoning

**Comment by pohly** [comment]
As I mentioned in https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner/pull/55#issuecomment-890904417, repo owner are not blocked and can update their copy of csi-release-tools.

**Comment by RinkiyaKeDad** [comment]
Unassinging myself for the moment, but please let me know if there's something that needs help :)

/unassign

**Comment by spiffxp** [comment]
Two more repos have crept up since this issue was created.  So I opened:
- https://github.com/kubernetes-sigs/lwkd/pull/117
- https://github.com/kubernetes-sigs/prometheus-adapter/pull/454

And poked for an update on:
- https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner/pull/55
- https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner/pull/117

**Comment by spiffxp** [comment]
Hopefully the last bit needed:
- opened https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner/pull/144 (based off of https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner/pull/70)

**Comment by spiffxp** [comment]
/close
And we're done!

Thanks for the help @RinkiyaKeDad @anubha-v-ardhan @pohly @wongma7 @ikarldasan @praveenghuge
